libcloud/compute/drivers/outscale.py

```diff
@@ -0,0 +1,1109 @@
+            data.update({"VmId": vm_id})
+        if allow_relink is not None:
+            data.update({"AllowRelink": allow_relink})
+        data = json.dumps(data)
+        signer = OSCRequestSignerAlgorithmV4(access_key=self.key,
+                                             access_secret=self.secret,
+                                             version=self.version,
+                                             connection=self.connection)
+        headers = signer.get_request_headers(action=action,
+                                             data=data,
+                                             service_name=self.service_name,
+                                             region=self.region)
+        endpoint = self._get_outscale_endpoint(self.region,
+                                               self.version,
+                                               action)
+        return requests.post(endpoint, data=data, headers=headers)
+
+    def detach_public_ip(self, public_ip=None,
+                         link_public_ip_id=None,
+                         dry_run=False):
+        """
+        Detach a volume.
+
+        :param      public_ip: (Required in a Net) The ID representing the
+        association of the EIP with the VM or the NIC
+        :type       public_ip: ``str``
+
+        :param      link_public_ip_id: (Required in a Net) The ID
+        representing the association of the EIP with the
+        VM or the NIC.
+        :type       link_public_ip_id: ``str``
+
+        :param      dry_run: If true, checks whether you have the required
+        permissions to perform the action.
+        :type       dry_run: ``bool``
+
+        :return: the attached volume
+        :rtype: ``dict``
+        """
+        action = "UnlinkPublicIp"
+        data = {"DryRun": dry_run}
+        if public_ip is not None:
+            data.update({"PublicIp": public_ip})
+        if link_public_ip_id is not None:
+            data.update({"LinkPublicIpId": link_public_ip_id})
+        data = json.dumps(data)
+        signer = OSCRequestSignerAlgorithmV4(access_key=self.key,
+                                             access_secret=self.secret,
+                                             version=self.version,
+                                             connection=self.connection)
+        headers = signer.get_request_headers(action=action,
+                                             data=data,
+                                             service_name=self.service_name,
+                                             region=self.region)
+        endpoint = self._get_outscale_endpoint(self.region,
+                                               self.version,
+                                               action)
+        return requests.post(endpoint, data=data, headers=headers)
+
+    def create_node(self,
+                    image_id,
+                    dry_run=False,
+                    block_device_mapping=None,
```

Comment:
Argument which are not part of the Libcloud standard API (aka are not defined on the base NodeDriver class) need to be prefixed with ``ex_`` - ex_bsu_optimized, ex_nics, etc.